### PR TITLE
feat: update the role of a project member

### LINKED ISSUE #760: Update the role of a project member
I want to change from admin to member (except for the last admin)

Image

### PR DESCRIPTION
**Pull request by YohannesTesfay**
Closes #760

### CODE REVIEW

File: apps/web/frontend/components/projectMemberList/projectMemberList.tsx [hunk 1]
```diff
@@ -60,7 +79,12 @@ export const ProjectMemberList = (props: ProjectMemberListProps) => {
                 {user.name}
               </TableCell>
               <TableCell>
-                <RoleLabel role={user.role} />
+                <RoleLabel
+                  role={user.role}
+                  loading={fetching}
+                  onUpgrade={() => handleUpdateProjectMembership(user.id, Role.Admin)}
+                  onDowngrade={() => handleUpdateProjectMembership(user.id, Role.Member)}
```
Comment (PascalHelbig) [inline apps/web/frontend/components/projectMemberList/projectMemberList.tsx:86]: I think up and downgrading should only be possible for admins

File: apps/web/frontend/components/projectMemberList/roleLabel.tsx [hunk 2]
```diff
@@ -3,12 +3,26 @@ import { Label } from './label'
...
-
-  return <Label color="yellow">Member</Label>
-}
+export const RoleLabel = ({ role, onUpgrade, onDowngrade, loading }: RoleLabelProps) => (
+  <div className="flex flex-row space-x-5">
+    <Label color={role === Role.Admin ? 'blue' : 'yellow'}>{role === Role.Admin ? 'Admin' : 'Member'}</Label>
+    <Label color={role === Role.Admin ? 'yellow' : 'blue'}>
+      <div className="flex flex-row space-x-2">
+        {role === Role.Admin ? (
+          <button onClick={onDowngrade} disabled={loading} className="hover:hover:scale-95 disabled:opacity-50">
```
Comment (PascalHelbig) [inline apps/web/frontend/components/projectMemberList/roleLabel.tsx:17]: `hover:hover:`...

File: apps/web/frontend/components/projectMemberList/label.tsx [hunk 3]
```diff
@@ -3,14 +3,15 @@ import { ReactNode } from 'react'
 interface LabelProps {
   children: ReactNode
   color: 'blue' | 'yellow'
+  invert?: boolean
```
[the remark this answers is not in the document]
Comment (moncapitaine) [reply apps/web/frontend/components/projectMemberList/label.tsx:6]: do not use Label for Buttons.

File: apps/web/frontend/components/projectMemberList/projectMemberList.tsx [hunk 4]
```diff
@@ -60,7 +79,12 @@ export const ProjectMemberList = (props: ProjectMemberListProps) => {
                 {user.name}
               </TableCell>
               <TableCell>
-                <RoleLabel role={user.role} />
+                <RoleLabel
```
Comment (moncapitaine) [inline apps/web/frontend/components/projectMemberList/projectMemberList.tsx:82]: make one for the label and another for the button to downgrade or upgrade. (Both buttons only usable for project admins)

Comment (moncapitaine) [reply apps/web/frontend/components/projectMemberList/projectMemberList.tsx:82]: do not allow to upgrade or downgrade yourself

**Review by moncapitaine — CHANGES_REQUESTED** [review]
please do the upgrade/downgrade as buttons (not rendering as label).

Add missing tests for upgrade, downgrade, nothing

**Pull request merged**